File: MainServer/src/main/java/DTO/FeedbackVideoStreamer.java
package DTO;

import java.io.File;
import java.util.List;
import java.util.Map;

/**
 * This class is for returning the a streaming path to the customer for view the feedbackVideo
 */
public class FeedbackVideoStreamer {

    private String path;
    //TODO need to add here the List of comments because we cant send them in with the streaming video
        private List<String> detectors;

    public FeedbackVideoStreamer(File file, List<String> detectors) {
        this.path = file.getPath();
        this.detectors = detectors;
    }

    public String getPath() {
        return this.path;
    }
}

File: client/lib/Domain/Feedback/FeedBackVideoStreamer.dart
class FeedbackVideoStreamer {

  final String path;
  List<String> detectors;

  FeedbackVideoStreamer(this.path, this.detectors);

  FeedbackVideoStreamer.fromJson(Map<String, dynamic> json)
      : path = json['path'],
        detectors = json['detectors'];

  String getPath() {
    return "/"+this.path.replaceAll("\\", "/");
  }

  static FeedbackVideoStreamer factory(Map map) {
    List<String> list = List<String>.from(map['detectors']);
    return new FeedbackVideoStreamer(map['path'], list);
  }

}